### DISCUSSION TITLE
Zigbee2MQTT can't connect to Zigbee Bridge Pro

### DISCUSSION THREAD

**Discussion by mcarson75** [arendst/Tasmota#17998] [Q&A] [answered]
I have a Sonoff Zigbee Bridge Pro that had been working fine with Zigbee2MQTT using Tasmota 12.3.  I went to upgrade it to 12.4 and must have done something wrong, because it wouldn't boot (the details aren't important for this) so I started from scratch, and I cannot get Zigbee2MQTT to connect again.  I think I must be missing something in my understanding.

I haven't found a good guide that gives every step necessary to flash Tasmota and get this working with HA/Zigbee2MQTT, so I have put together steps from several guides.  Here are the steps I followed this latest time.  (I have tried modifying the order of these, setting the template manually, and several other variations with no luck.)  This was working fine just yesterday, so I really don't know what I am missing!  This does not appear to be a network issue.  The IP address/subnet did not change, and you can see in my steps I check connectivity and it seems right.

The piece I do not understand is how the coordinator fits into this?  I flash coordinator, but when I have it configured as `Sonoff ZBPro TCP` I do not see any messages showing that the coordinator has started on boot.  I don't know if this is correct or not.  If I disable autoconfig and reflash coordinator, then I get the "Permit Join All" and other button on the main screen, and the log messages in the console.  However, the TCP server will not start in this case.

But since the network is correct, the problem seems to be the TCP Server.  The port is open but no one is home, it seems.  I don't know what is supposed to be there or how to troubleshoot.  Can anyone give me some pointers?  I am completely stuck now.  Here are the steps I followed this time:

- Serial flash `tasmota32.bin` using ESP-Flasher.exe.  I see success, although no logs come up after this.
- Disconnect from serial, connect to power, find AP, configure network
- Connect to IP address
- OTA flash `tasmota32-zbbrdgpro.bin` (this step from [here](https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/))
- set up MQTT, backup Wifi, hostname
- Check fs size, is only 320
- Download partition wizard (this step from [here](https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/))
- Reboot
- Go into Partition wizard, migrate to Safeboot
- Resize fs to 1344 (from [here](https://tasmota.github.io/docs/Safeboot/#introducing-safeboot))
- Go into Manage file system and upload the following files:
  - `cc2652_flasher.be`
  - `intelhex.be`
  - `sonoff_zb_pro_flasher.be`
  - `sonoffzbpro_coord_20220219.hex`
- Apply auto-configuration "Sonoff ZBPro TCP" (step from [here](https://tasmota.github.io/docs/Safeboot/#introducing-safeboot))'
- Check and flash coordinator, flash completes successfully (this step from [here](https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/))
```
	18:52:44.567 FLH: Starting verification of HEX file
	18:52:54.832 FLH: Verification of HEX file OK
	18:53:08.951 FLH: cc2652_flasher rx=23 tx=19 rst=15 bsl=22
	18:53:08.998 FLH: Flashing started (takes 5 minutes during which Tasmota is unresponsive)
	18:56:55.808 FLH: Flashing completed: OK
	18:56:55.916 FLH: Flash crc32 0x000000 - 0x2FFFF = bytes('1598929A')
```
- Reboot - here are console logs
```
	00:00:02.373 HTP: Web server active on zigbeeprobridge with IP address 10.0.49.100
	18:58:58.723 MQT: Attempting connection...
	18:58:59.104 MQT: Connected
	18:58:59.109 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/tele/LWT = Online (retained)
	18:58:59.111 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/cmnd/POWER = 
	18:58:59.129 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/tele/INFO1 = {"Info1":{"Module":"TCP ZBBridge Pro","Version":"12.4.0(zbbrdgpro)","FallbackTopic":"cmnd/DVES_6071DC_fb/","GroupTopic":"tasmota/downstairs/tasmotas/zigbee/cmnd/"}}
	18:58:59.146 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/tele/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"zigbeeprobridge","IPAddress":"10.0.49.100","IP6Global":"","IP6Local":"fe80::4222:d8ff:fe60:71dc"}}
	18:58:59.174 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/tele/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":11}}
	18:58:59.208 RUL: SYSTEM#BOOT performs "TCPStart 8888"
	18:58:59.211 TCP: Starting TCP server on port 8888
	18:58:59.215 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/RESULT = {"TCPStart":"Done"}
	18:59:00.282 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"https://ota.tasmota.com/tasmota32/tasmota32-zbbrdgpro.bin","RestartReason":"Software reset CPU","Uptime":"0T00:00:05","StartupUTC":"2023-02-19T17:58:55","Sleep":50,"CfgHolder":4617,"BootCount":10,"BCResetTime":"2023-02-19T18:20:00","SaveCount":28}}
	18:59:00.335 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS11 = {"StatusSTS":{"Time":"2023-02-19T18:59:00","Uptime":"0T00:00:05","UptimeSec":5,"Heap":161,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":38,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":47},"Wifi":{"AP":1,"SSId":"hokie_iot","BSSId":"76:83:C2:94:32:B2","Channel":1,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:02"}}}
	18:59:02.789 QPC: Reset
	18:59:03.823 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/tele/STATE = {"Time":"2023-02-19T18:59:03","Uptime":"0T00:00:08","UptimeSec":8,"Heap":162,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":81,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":47},"Wifi":{"AP":1,"SSId":"hokie_iot","BSSId":"76:83:C2:94:32:B2","Channel":1,"Mode":"11n","RSSI":66,"Signal":-67,"LinkCount":1,"Downtime":"0T00:00:02"}}
	18:59:23.962 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"zigbeebridgepro","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
	18:59:24.001 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"https://ota.tasmota.com/tasmota32/tasmota32-zbbrdgpro.bin","RestartReason":"Software reset CPU","Uptime":"0T00:00:28","StartupUTC":"2023-02-19T17:58:55","Sleep":50,"CfgHolder":4617,"BootCount":11,"BCResetTime":"2023-02-19T18:20:00","SaveCount":29}}
	18:59:24.041 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS2 = {"StatusFWR":{"Version":"12.4.0(zbbrdgpro)","BuildDateTime":"2023-02-18T16:10:03","Core":"2_0_6","SDK":"4.4.3.20230129","CpuFrequency":160,"Hardware":"ESP32-D0WD-V3 rev.3","CR":"489/699"}}
	18:59:24.070 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["hokie_iot","hokie_ap"],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
	18:59:24.103 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS4 = {"StatusMEM":{"ProgramSize":1460,"Free":395,"Heap":158,"StackLowMark":3,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DIO","Features":["00000809","8F8A87CF","00548001","000000CF","010013C0","C0000981","40024080","00200000","5400082C","00000080"],"Drivers":"1,2,3,4,5,8,9,10,12,16,20,21,23,24,27,29,35,38,41,50,52,59,62,82","Sensors":"1,2,3,5,6,127"}}
	18:59:24.155 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS5 = {"StatusNET":{"Hostname":"zigbeeprobridge","IPAddress":"10.0.49.100","Gateway":"10.0.50.1","Subnetmask":"255.255.252.0","DNSServer1":"10.0.50.1","DNSServer2":"0.0.0.0","Mac":"40:22:D8:60:71:DC","IP6Global":"","IP6Local":"fe80::4222:d8ff:fe60:71dc","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"10.0.50.1","DNSServer2":"0.0.0.0","Mac":"00:00:00:00:00:00","IP6Global":"","IP6Local":""},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
	18:59:24.207 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS6 = {"StatusMQT":{"MqttHost":"ha.iot.carson-family.com","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_6071DC","MqttUser":"matt","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
	18:59:24.240 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS7 = {"StatusTIM":{"UTC":"2023-02-19T17:59:24","Local":"2023-02-19T18:59:24","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"07:51","Sunset":"18:16"}}
	18:59:24.275 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS10 = {"StatusSNS":{"Time":"2023-02-19T18:59:24"}}
	18:59:24.288 MQT: tasmota/downstairs/zigbeebridgepro/zigbee/stat/STATUS11 = {"StatusSTS":{"Time":"2023-02-19T18:59:24","Uptime":"0T00:00:29","UptimeSec":29,"Heap":153,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":47},"Wifi":{"AP":1,"SSId":"hokie_iot","BSSId":"76:83:C2:94:32:B2","Channel":1,"Mode":"11n","RSSI":62,"Signal":-69,"LinkCount":1,"Downtime":"0T00:00:02"}}}
```
- Note that TCP server has started
- Test via netcat from Home Assistant console
```
	[core-ssh config]$ nc -z -v 10.0.49.100 8888
	10.0.49.100 (10.0.49.100:8888) open
```
Acknowledgement in Tasmota console:
```
	19:00:26.631 TCP: Got connection from 10.0.50.30
```
- Test via netcat from another machine on subnet:
```
	$ nc -z -v 10.0.49.100 8888
	Connection to 10.0.49.100 8888 port [tcp/*] succeeded!
```
Acknowledgement in Tasmota console:
```
	19:00:32.141 TCP: Got connection from 10.0.50.15
```
- Check config again in Zigbee2MQTT:
```
	port: tcp://10.0.49.100:8888
	adapter: zstack
```
- Start Zigbee2MQTT, get failure:
Acknowledgement of connection in Tasmota:
```
	19:00:56.419 TCP: Got connection from 10.0.50.30
```
Failure logs in Zigbee2MQTT:
```
	[13:00:53] INFO: Preparing to start...
	[13:00:53] INFO: Socat not enabled
	[13:00:54] INFO: Starting Zigbee2MQTT...
	Zigbee2MQTT:info  2023-02-19 13:00:57: Logging to console and directory: '/config/zigbee2mqtt/log/2023-02-19.13-00-57' filename: log.txt
	Zigbee2MQTT:info  2023-02-19 13:00:57: Starting Zigbee2MQTT version 1.30.1 (commit #unknown)
	Zigbee2MQTT:info  2023-02-19 13:00:57: Starting zigbee-herdsman (0.14.89)
	Zigbee2MQTT:error 2023-02-19 13:01:16: Error while starting zigbee-herdsman
	Zigbee2MQTT:error 2023-02-19 13:01:16: Failed to start zigbee
	Zigbee2MQTT:error 2023-02-19 13:01:16: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
	Zigbee2MQTT:error 2023-02-19 13:01:16: Exiting...
	Zigbee2MQTT:error 2023-02-19 13:01:16: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
		at ZStackAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:103:27)
		at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
		at Zigbee.start (/app/lib/zigbee.ts:58:27)
		at Controller.start (/app/lib/controller.ts:101:27)
		at start (/app/index.js:107:5)
```

Note that I have also attempted to set up with ZHA, and get the same result - can't connect to socket.

I'm at my wit's end here.  I've tried everything I can think of in the last 24 hours and nothing has made a bit of difference.  It always fails in the same way.  What am I missing?

One other note - I did notice that the templates [here](https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/) and [here](https://zigbee.blakadder.com/Sonoff_ZBBridge-P.html) are different - the TCP RX and TX are reversed from each other.  However, I have tried it both ways and I still can connect via nc with both templates and Z2M still fails with both templates.

**Comment by mcarson75** [comment, marked as answer]
Update: I got this working after finding [https://github.com/arendst/Tasmota/discussions/17213](https://github.com/arendst/Tasmota/discussions/17213).  I mapped GPIO 15 to a relay and reset it and now I am able to connect.

I then set
```
poweronstate 5
pulsetime1 10
rule1 on system#boot do TCPStart 8888 endon on system#boot do power1 toggle endon
```

so that this resets on boot.

**Reply by OhmegaStar** [reply]
thanks a lot for this post. I faced same problem as you, but from a new installation, with the latest version with autoconfig and updated coordinator available.
My only step missing was the last, to add the poweron / power toggle rule. After a soft reboot my zigbee2mqtt was finally able to connect.

**Reply by OhmegaStar** [reply]
A bit Odd, Im seeing some issues with the power1 toggle command;

`11:04:17.950 MQT: tele/tasmota_0DD8EC/INFO1 = {"Info1":{"Module":"TCP ZBBridge Pro","Version":"13.4.0.2(27ccf04-zbbrdgpro)","FallbackTopic":"cmnd/DVES_0DD8EC_fb/","GroupTopic":"cmnd/tasmotas/"}}
11:04:17.966 MQT: tele/tasmota_0DD8EC/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-0DD8EC-6380","IPAddress":"192.168.0.179","IP6Global":"","IP6Local":"fe80::redacted%st1"}}
11:04:17.981 MQT: tele/tasmota_0DD8EC/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":55}}
11:04:17.998 TFS: File 'autoexec.bat' not found
11:04:18.008 RUL: SYSTEM#BOOT performs "TCPStart 8888"
11:04:18.009 SRC: Rule
11:04:18.010 CMD: Grp 0, Cmd 'TCPSTART', Idx 1, Len 4, Pld 8888, Data '8888'
11:04:18.011 TCP: Starting TCP server on port 8888
11:04:18.016 MQT: stat/tasmota_0DD8EC/RESULT = {"TCPStart":"Done"}
11:04:18.022 RUL: SYSTEM#BOOT performs "power1 toggle"
11:04:18.023 SRC: Rule
11:04:18.025 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 6, Pld 2, Data 'toggle'
11:04:18.027 MQT: stat/tasmota_0DD8EC/RESULT = {"Command":"Error"}
11:04:20.614 WIF: IPv6 Global redacted
11:04:20.908 HTP: Main Menu`

My GPIO15 Config:
![image](https://github.com/arendst/Tasmota/assets/15914993/1596c68a-82a5-46b5-aa55-8d97c1fcea67)

Causing issues on restart for z2m

Cant Really see whats wrong with the config.

Any suggestions?

/OhmegaStar

**Reply by sfromis** [reply]
You did not follow the suggestion of mcarson75, which was to instead config gpio 15 as a relay. Of course, this can only be workable if you make sure to have done all 3 suggested commands.

**Comment by andreapede** [comment]
> Update: I got this working after finding [https://github.com/arendst/Tasmota/discussions/17213](https://github.com/arendst/Tasmota/discussions/17213).  I mapped GPIO 15 to a relay and reset it and now I am able to connect.
> 
> I then set
> ```
> poweronstate 5
> pulsetime1 10
> rule1 on system#boot do TCPStart 8888 endon on system#boot do power1 toggle endon
> ```
> 
> so that this resets on boot.

I would thank you because after I struggled for days over forums the solution worked correctly.
It was evident enough to me that the normal restart of the bridge didn't work. Only unplugging the power supply could restart the zigbee operation.
Gpio 15 as relay made the difference. Ty

**Reply by OhmegaStar** [reply]
Can you show what is meant by "configure gpio 15 as relay", im still uncertain about this step (so it's probably obvious....)??
Tia.
/OhmegaStar

**Reply by mcarson75** [reply]
In your screenshot above from Apr 2 you have GPIO 15 configured as `Zigbee Rst 1` under "Configure Template".  This needs to be `Relay 1`.

![image](https://github.com/user-attachments/assets/f3956edd-e892-4664-b758-93df4ade5993)

**Reply by OhmegaStar** [reply]
Tanks for the clear explanation. Will get this fixed 👌👍

**Comment by Igel82** [comment]
Hello everyone, I have the same problem. But I don't know where to insert the text after I have configured GPIO 15 as relay. Where must this text be written in zigbee2mqtt? 
```
poweronstate 5
pulsetime1 10
rule1 on system#boot do TCPStart 8888 endon on system#boot do power1 toggle endon
```

**Reply by mcarson75** [reply]
These are entered in the Tasmota console.

**Reply by Igel82** [reply]
I have entered it here and restarted it, but unfortunately it still does not work.
![grafik](https://github.com/user-attachments/assets/b2e3ca7d-49f7-4253-a757-b90d29195abc)
is this setting correct?
![grafik](https://github.com/user-attachments/assets/21cf9b3b-3abb-468d-8a7d-062302e848dc)

**Reply by mcarson75** [reply]
You need to enable rule 1 with
```
rule1 1
```

**Reply by Igel82** [reply]
I did that, but unfortunately I still get this error message
![grafik](https://github.com/user-attachments/assets/34660153-c0d3-4ff2-9601-37cdf033ed00)
![grafik](https://github.com/user-attachments/assets/7e1ec950-5675-4a63-a5ac-b6d4bcf1ff08)

**Reply by sfromis** [reply]
You should've restarted again, as the two rules can only trigger at boot.

**Reply by mcarson75** [reply]
It's honestly been a long time since I set this up so I don't remember the details about why but I also have `adapter: zstack`
![image](https://github.com/user-attachments/assets/79b20279-43d2-437b-8007-f42a964772ef)

**Reply by Igel82** [reply]
I have restarted home assistant and the bridge but unfortunately the same error still occurs. I have also added the adapter
after entering into the console, the green led lights up permanently, is that correct?

**Reply by mcarson75** [reply]
Do you happen to have a firewall between home assistant and your Zigbee bridge?

**Reply by Igel82** [reply]
no there is no firewall in between. do i have to enter this in the terminal of the home assistant to test it? 
[core-ssh config]$ nc -z -v 10.0.49.100 8888
10.0.49.100 (10.0.49.100:8888) open
![grafik](https://github.com/user-attachments/assets/7fcfae58-094a-4893-8fab-b74d6330ff24)
should I try ZHA?

**Reply by mcarson75** [reply]
What do you see in the Tasmota console if you run
```
TCPStart 8888
```
?

After you test that restart Zigbee2MQTT.

**Reply by Igel82** [reply]
![grafik](https://github.com/user-attachments/assets/c625f4e3-3ec2-4b9d-b0d9-6d773af85646)
I have restarted Zigbee2MQTT but it does not work. If I start Zigbee2MQTT without the serial adpater, do I get the same error or is it mandatory for Zigbee2MQTT to start?

**Reply by sfromis** [reply]
When you want TCP instead of local Zigbee, you need to configure the template, replacing components `Zigbee Tx/Rx` with `TCP Tx/Rx`

**Reply by Igel82** [reply]
do you mean this configuration?
![grafik](https://github.com/user-attachments/assets/ffab0408-039b-4f1b-915d-b55116ad3ab5)
or can i do this over the auto configuration
![grafik](https://github.com/user-attachments/assets/09bfe21c-dda4-49a9-8030-00d4893cdec5)

**Reply by sfromis** [reply]
What I meant was to *change* to gpio components in the dropdowns, but it does indeed look like you can get to the same using the autoconf entry you indicate there, with the same end result.

**Reply by Igel82** [reply]
Thank you very much, it worked. I was able to connect now, but it doesn't show up as a device in zigbee2mqtt?
![386638831-ae4fa18d-0d91-462b-9d61-4bc977fdb041](https://github.com/user-attachments/assets/12f973cf-d3df-46ab-a306-8b01edb51214)

**Reply by mcarson75** [reply]
Not sure exactly what you expect to see, but the bridge itself is not a Zigbee device.  You should be able to manage things in the Zigbee2MQTT web ui now (click on "Open Wb UI on the addon).

**Reply by Igel82** [reply]
sorry my mistake, only taught-in devices such as sensors etc. are displayed, right?

**Reply by mcarson75** [reply]
Yes, in the web ui you can click on "Permit Join" and then click the join button on your Zigbee device and it should show up in the UI.

**Reply by Igel82** [reply]
Thanks guys I have now been able to integrate my motion detector and I can also control it via automations in Home Assistant

**Reply by Igel82** [reply]
can you tell me if it is possible to connect several such bridges with zigbee2mqtt?

**Reply by OhmegaStar** [reply]
Hi, No only one bridge per zigbee network. For more than one network you cloud enable zhm and add a second bridge there.

**Reply by sfromis** [reply]
If you want to have two separate Zigbee networks, each with their own coordinator and keys, you could have two copies of Zigbee2MQTT running.

**Reply by Igel82** [reply]
oh okay, thanks for the info. Is there a way to increase the range of the bridge with a repeater or something?
something like this, for example [Repeater](https://www.zigbee2mqtt.io/devices/WG001.html)

**Reply by sfromis** [reply]
Most plugged-in Zigbee devices (not battery-powered), like plugs, should already work as Zigbee router nodes, without using a separate box for this.

**Comment by Sr-psycho** [comment]
Yes there is a [way to force reset](https://github.com/arendst/Tasmota/discussions/17213#discussioncomment-4276569) on the MCU, driving GPIO15 low. You can assigne a Relay on GPIO15 and turn it OFF then ON again to force a hardware reset.
Zigbee2MQTT could also send a software reset by sending a sequence of 256x 0xFF.

**Comment by tequilaj-gh** [comment]
I faced the same issues (Tasmota 14.2.0, sonoffzbpro_coord_20220219.hex) and the extra steps described here did work. 

Thank you very much.
